### PR TITLE
[MRG+1] update Scrapy to use parsel 1.5

### PR DESCRIPTION
**Pull request by kmike**
Fixes https://github.com/scrapy/scrapy/issues/3317.

Todo:

* [x] update Selector and SelectorList API documentation
* [x] revise Selectors tutorial: either bring more from the parsel's tutorial, or link to missing pieces (like has-class XPath extension); maybe rearrange some of the sections.
* [x] check .extract usages
* [x] find more places where .attrib makes code simpler
* [x] update after #3400.
* [x] fix "Removing namespaces" example - see https://github.com/scrapy/parsel/pull/119

### CODE REVIEW

**Comment by stummjr** [comment]
@kmike just a heads-up that I started reviewing this today.

**Comment by eliasdorneles** [comment]
I love this PR so much <3 =)

File: tests/test_utils_iterators.py [hunk 1]
```diff
@@ -30,10 +30,13 @@ def test_xmliter(self):
         response = XmlResponse(url="http://example.com", body=body)
         attrs = []
         for x in self.xmliter(response, 'product'):
-            attrs.append((x.xpath("@id").extract(), x.xpath("name/text()").extract(), x.xpath("./type/text()").extract()))
+            attrs.append((
+                x.attrib['id'],
+                x.xpath("name/text()").extract(),
+                x.xpath("./type/text()").extract()))
```
Comment (stummjr) [inline tests/test_utils_iterators.py]: `.getall()` here too?

File: docs/intro/tutorial.rst [hunk 2]
```diff
@@ -641,7 +649,7 @@ this time for scraping author information::
 
         def parse_author(self, response):
             def extract_with_css(query):
-                return response.css(query).extract_first().strip()
+                return response.css(query).get().strip()
```
[the remark this answers is not in the document]
Comment (kmike) [reply docs/intro/tutorial.rst]: TIL response.css('foo::text').get() returns None if foo is found, but there is no text inside.

**Comment by kmike** [comment]
Thanks for a careful review @stummjr! How does it look now?

**Pull request merged**